ratpack-kafka-consumer/src/main/java/smartthings/ratpack/kafka/Consumer.java

```diff
@@ -0,0 +1,57 @@
+package smartthings.ratpack.kafka;
+
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import ratpack.func.Action;
+
+import java.util.Properties;
+
+/**
+ * Interface for creating a Kafka Consumer.  Implementers will have their consume method called as messages become
+ * available.
+ * @param <K> Key deserialization class.
+ * @param <V> Value deserialization class.
+ */
+public interface Consumer<K, V> {
+	/**
+	 * Method will be called on successful Kafka poll with fetched messages.
+	 * @param records
+	 */
+	void consume(ConsumerRecords<K, V> records) throws Exception;
+
+	/**
+	 * Kafka topics in which to subscribe.
+	 * @return
+	 */
+	String[] getTopics();
+
+	/**
+	 * Kafka Consumer Group ID.
+	 * @return
+	 */
+	String getGroup();
+
+	/**
+	 * Determines how many consumers will be spun up on application start.
+	 * @return
+     */
+	default Integer getConcurrencyLevel() {
```

Comment:
I'd recommend against this option.  It should be one unless `topics` or `group` are changing with it.  Otherwise if misused this will cause the partitions to be all assigned to a single process and prevent scaling as expected when adding more processes in.  It also will create unnecessary coordination between consumers in the same process.